pkg/operator/ceph/object/spec.go

```diff
@@ -453,6 +458,24 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 		container.Args = append(container.Args, hostingOptions)
 	}
 
+	// set RGW read Affinity
+	isCrushLocationSet := false
+	if c.store.Spec.Gateway.ReadAffinity != nil {
+		if c.clusterInfo.CephVersion.IsAtLeastTentacle() {
+			container.Args = append(container.Args, cephconfig.NewFlag(radosReadReplicaPolicy, c.store.Spec.Gateway.ReadAffinity.Type))
+			if c.store.Spec.Gateway.ReadAffinity.Type == localizedReadReplicaPolicy {
+				container.Command = []string{"bash", "-x", "-c", `exec radosgw --crush-location="host=${NODE_NAME//./-}" "$@"`}
+				isCrushLocationSet = true
+			}
```

Comment:
Let's not require a specific replica policy to set the crush location and instead just base this on whether read affinity is nil or non-nil. 

Same reasoning as here -- we can support new options by only patching the CRD instead of having to change golang code: https://github.com/rook/rook/pull/15588#discussion_r2014856295

```suggestion
			container.Args = append(container.Args, cephconfig.NewFlag(radosReadReplicaPolicy, c.store.Spec.Gateway.ReadAffinity.Type))
			container.Command = []string{"bash", "-x", "-c", `exec radosgw --crush-location="host=${NODE_NAME//./-}" "$@"`}
			isCrushLocationSet = true
```

Alternately, we could change the logic to apply `crush-location` if the value is not 'default' or 'balanced'. That would also allow us to support new values in the future without code work.

Reply:
Updated the code to set `crush-location` when readAffinity is enabled. 

> Let's not require a specific replica policy to set the crush location and instead 
As per ceph folks, we only need crush location for localized reads. But adding crush location for other read policies should not be an issue. Was trying to restrict the usage of the bash script to a minimum.